src/Types/Type.php

```diff
@@ -48,13 +48,6 @@ abstract class Type
 
     private static ?TypeRegistry $typeRegistry = null;
 
-    /**
-     * @internal Do not instantiate directly - use {@see Type::addType()} method instead.
```

Comment:
A type registered through `addType` is still required to have no arguments in the constructor. We still need a check for that.

Reply:
It looks like we need to deprecate the static `Type` methods that accept `string $className` in favor of `TypeRegistry` and the corresponding counterparts which accept `Type $type`.

Only after the deprecated methods have been removed, this constraint could be dropped.

Reply:
See https://github.com/doctrine/dbal/pull/5530